src/nodes.coffee

```diff
@@ -2229,8 +2229,11 @@ exports.Assign = class Assign extends Base
             nestedProperties = prop.value.variable.base.properties
             [prop.value.value, nestedSourceDefault] = prop.value.value.cache o
           if nestedProperties
-            nestedSource = new Value source.base, source.properties.concat [new Access getPropKey prop]
-            nestedSource = new Value new Op '?', nestedSource, nestedSourceDefault if nestedSourceDefault
+            if source.properties
```

Comment:
I guess it's just not clear to me why we need to check if `source.properties` is set. I imagine this is roughly equivalent to `source instanceof Value`, and my 2 questions are:

1. Would it be easier to ensure `source` is a `Value` (e.g. in the grammar/wherever `source` is created)?
1. If `source` is *not* a value (e.g. `not source.properties`) then we do not execute the line where we concat the property key. Why does this work?

I'll try and find some time to poke around to answer my own questions, since it's probably just my own misunderstandings. Since the tests work etc. this can probably go in, and if I find any issues I can submit a new ticket/fix.